addons/mail/static/src/models/use_drag_visible_drop_zone.js

```diff
@@ -0,0 +1,96 @@
+            document.addEventListener('dragleave', this._onDragleaveListener, true);
+            document.addEventListener('drop', this._onDropListener, true);
+
+            // Thoses Events prevent the browser to open or download the file if
+            // it's dropped outside of the dropzone
+            window.addEventListener('dragover', ev => ev.preventDefault());
+            window.addEventListener('drop', ev => ev.preventDefault());
+        },
+        _willDelete() {
+            document.removeEventListener('dragenter', this._onDragenterListener, true);
+            document.removeEventListener('dragleave', this._onDragleaveListener, true);
+            document.removeEventListener('drop', this._onDropListener, true);
+
+            window.removeEventListener('dragover', ev => ev.preventDefault());
+            window.removeEventListener('drop', ev => ev.preventDefault());
+        },
+    },
+    recordMethods: {
+        /**
+         * Shows the dropzone when entering the browser window, to let the user know
+         * where he can drop its file.
+         * Avoids changing state when entering inner dropzones.
+         *
+         * @private
+         * @param {DragEvent} ev
+         */
+        _onDragenterListener(ev) {
+            if (
+                this.dragCount === 0 &&
+                ev.dataTransfer &&
+                ev.dataTransfer.types.includes('Files')
+            ) {
+                this.update({ isVisible: true });
+            }
+            this.update({ dragCount: increment() });
+        },
+        /**
+         * @private
+         * @param {DragEvent} ev
+         */
+        _onDragleaveListener(ev) {
+            this.update({ dragCount: decrement() });
+            if (this.dragCount === 0) {
+                this.update({ isVisible: false });
+            }
+        },
+        /**
+         * @private
+         * @param {DragEvent} ev
+         */
+        _onDropListener(ev) {
+            this.update({
+                dragCount: 0,
+                isVisible: false,
+            });
+        },
+    },
+    fields: {
+        attachmentBoxViewOwner: one('AttachmentBoxView', {
+            inverse: 'useDragVisibleDropZone',
+            readonly: true,
+        }),
+        composerViewOwner: one('ComposerView', {
+            inverse: 'useDragVisibleDropZone',
+            readonly: true,
+        }),
+        /**
+         * Counts how many drag enter/leave happened globally. This is the only
+         * way to know if a file has been dragged out of the browser window.
+         */
+        dragCount: attr({
+            default: 0
+        }),
+        /**
+         * Determine whether the drop zone should be visible or not.
+         * Note that this is an observed value, and primitive types such as
+         * boolean cannot be observed, hence this is an object with boolean
+         * value accessible from `.value`
+         */
```

Comment:
Outdated comment :)
I think it safe to remove it now that it's a regular field.

Reply:
I just kept the 1st sentence, which is still relevant.